CDT/conanfile.py

```diff
@@ -3,54 +3,75 @@
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 from conan import ConanFile
-from conan.tools.cmake import CMake, CMakeToolchain
-from conan.tools.files import collect_libs
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout, CMakeDeps
+from conan.tools.build import valid_min_cppstd
 
 
 class CDTConan(ConanFile):
     name = "cdt"
-    version = "1.3.0"
+    version = "1.4.4"
     license = "MPL-2.0 License"
     url = "https://github.com/artem-ogre/CDT"
     description = "Numerically robust C++ implementation of constrained Delaunay triangulation (CDT)"
     settings = "os", "compiler", "build_type", "arch"
-    generators = "CMakeDeps"
     options = {
         "shared": [True, False],
-        "use_boost": [True, False],
-        "as_compiled_library": [True, False],
-        "enable_testing": [True, False],
+        "fPIC": [True, False],
+        "header_only": [True, False],
         "enable_callback_handler": [True, False],
+        "enable_testing": [True, False],
     }
     default_options = {
         "shared": False,
-        "use_boost": False,
-        "as_compiled_library": False,
-        "enable_testing": False,
+        "fPIC": True,
+        "header_only": False,
         "enable_callback_handler": False,
+        "enable_testing": False,
     }
     exports_sources = "*", "!.idea", "!conanfile.py"
 
+    @property
+    def _needs_boost(self):
+        #Versions below C++11 require boost
+        return not valid_min_cppstd(self, "11")
+    
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+    
+    def configure(self):
+        if self._needs_boost:
+            self.options["boost"].header_only = True
+
+        if self.options.get_safe("shared") or self.options.header_only:
+            self.options.rm_safe("fPIC")
+        if self.options.header_only:
+            self.options.rm_safe("shared")
+
     def requirements(self):
-        if self.options.use_boost:
-            self.requires("boost/1.83.0")
+        if self._needs_boost:
+            self.requires("boost/[^1.83.0]")        
```

Comment:
I only used it for local development, so 1.83 was the most recent at that point. I guess it could be bumped down to the oldest boost available in conan center.